### PR TITLE
fix: use SlotSize as VDisk size limit if AvailableSize is 0

### LINKED ISSUE #2993: Display VDisk usage based on SlotSize if AvailableSize is 0

### PR DESCRIPTION
**Pull request by artemmufazalov**
Closes #2993 

Examples:

Storage: https://nda.ya.ru/t/Ct51L8tA7LkAzN
Group: https://nda.ya.ru/t/qivlXwJ97LkAwv
VDisk: https://nda.ya.ru/t/k4mB7qyU7LkAuX

<h2>Greptile Overview</h2>

Updated On: 2025-10-22 12:38:57 UTC

<h3>Summary</h3>

This PR fixes VDisk usage display when `AvailableSize` is 0 by using `SlotSize` as the size limit instead. Previously, VDisks would show `AllocatedSize + AvailableSize` as the total, which resulted in incorrect 100% usage display when `AvailableSize` was 0, even if the disk hadn't reached its actual slot size limit.

**Key Changes:**
- Renamed `TotalSize` to `SizeLimit` in VDisk types to better reflect its purpose
- Modified `prepareVDiskSizeFields()` to use `SlotSize` as the limit when `AvailableSize` is 0 or undefined
- Updated `DiskStateProgressBar` to clamp percentage values (can now exceed 100% when allocated exceeds slot size)
- Added logic in PDisk utils to adjust slot total for visualization when used size exceeds limit
- Refactored VDisk preparation to use consistent `prepareGroupsVDisk()` function
- Comprehensive test coverage for all edge cases including overflow scenarios

<h3>Confidence Score: 5/5</h3>

- This PR is safe to merge with no risk. The changes are well-tested, logically sound, and fix a legitimate bug.
- The implementation is thorough with comprehensive test coverage for all edge cases. The logic correctly handles the scenario where AvailableSize is 0 by falling back to SlotSize. The refactoring improves code consistency, and the progress bar clamping prevents visual overflow issues. All test cases validate the expected behavior.
- No files require special attention

<h3>Important Files Changed</h3>

File Analysis

| Filename | Score | Overview |
|----------|-------|----------|
| src/utils/disks/prepareDisks.ts | 5/5 | Core logic change to use SlotSize as size limit when AvailableSize is 0. Logic is correct and well-tested. |
| src/components/VDiskInfo/VDiskInfo.tsx | 5/5 | UI updated to use SizeLimit instead of calculating from AllocatedSize + AvailableSize. Correct implementation. |
| src/components/DiskStateProgressBar/DiskStateProgressBar.tsx | 5/5 | Added clamping for diskAllocatedPercent to prevent visual overflow when usage exceeds 100%. Good defensive programming. |
| src/store/reducers/pdisk/utils.ts | 5/5 | Added logic to adjust slot Total when used exceeds limit for proper visualization sizing. Well-commented and tested. |

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant API as Backend API
    participant Prep as prepareVDiskSizeFields
    participant PDisk as preparePDiskDataResponse
    participant VDiskUI as VDiskInfo Component
    participant ProgBar as DiskStateProgressBar

    API->>Prep: VDisk data (AllocatedSize, AvailableSize, SlotSize)
    
    alt AvailableSize > 0
        Prep->>Prep: sizeLimit = allocated + available
    else AvailableSize = 0 and SlotSize exists
        Prep->>Prep: sizeLimit = slotSize
    else AvailableSize = 0 and no SlotSize
        Prep->>Prep: sizeLimit = allocated
    end
    
    Prep->>Prep: allocatedPercent = (allocated * 100) / sizeLimit
    Prep-->>PDisk: PreparedVDisk with SizeLimit & AllocatedPercent
    
    PDisk->>PDisk: Create slot items for visualization
    
    alt used > sizeLimit
        PDisk->>PDisk: Adjust total = used (for relative sizing)
    else used <= sizeLimit
        PDisk->>PDisk: total = sizeLimit
    end
    
    PDisk-->>VDiskUI: Slot data with adjusted totals
    VDiskUI->>VDiskUI: Display size as ProgressViewer(allocated, SizeLimit)
    
    PDisk-->>ProgBar: diskAllocatedPercent
    
    alt diskAllocatedPercent > 100
        ProgBar->>ProgBar: Clamp fillWidth to 100% (inverted: clamp to 0%)
    else diskAllocatedPercent <= 100
        ProgBar->>ProgBar: Use diskAllocatedPercent as-is
    end
    
    ProgBar->>ProgBar: Render progress bar with clamped width
```

### Test Status: <span style="color: orange;">⚠️ FLAKY</span>
  📊 [Full Report](https://ydb-platform.github.io/ydb-embedded-ui/2996/)

  | Total | Passed | Failed | Flaky | Skipped |
  |:-----:|:------:|:------:|:-----:|:-------:|
  | 378 | 375 | 0 | 1 | 2 |

  
  <details>
  <summary>Test Changes Summary ⏭️2 </summary>

  #### ⏭️ Skipped Tests (2)
1. Scroll to row, get shareable link, navigate to URL and verify row is scrolled into view (tenant/diagnostics/tabs/queries.test.ts)
2. Copy result button copies to clipboard (tenant/queryEditor/queryEditor.test.ts)

  </details>

  ### Bundle Size: ✅
  Current: 45.87 MB | Main: 45.87 MB
  Diff: +0.46 KB (0.00%)

  ✅ Bundle size unchanged.

  <details>
  <summary>ℹ️ CI Information</summary>

  - Test recordings for failed tests are available in the full report.
  - Bundle size is measured for the entire 'dist' directory.
  - 📊 indicates links to detailed reports.
  - 🔺 indicates increase, 🔽 decrease, and ✅ no change in bundle size.
  </details>

### CODE REVIEW

File: src/utils/disks/types.ts [hunk 1]
```diff
@@ -28,9 +28,9 @@ export interface PreparedVDisk
     StringifiedId?: string;
 
     AvailableSize?: number;
-    TotalSize?: number;
```
[the remark this answers is not in the document]
Comment (astandrik) [reply src/utils/disks/types.ts]: @greptile could you answer this question?
